src/localization/online_localizer/online_localizer.h

```diff
@@ -41,68 +41,67 @@
-    bool isReady() const;
-    void run();
-    void processImage(int quId);
-    // core working function
-    void matchImage(int quId);
-    std::vector<PathElement> getCurrentPath() const;
-    std::vector<PathElement> getLastNmatches(int N) const;
-
-    void writeOutExpanded(const std::string &filename) const;
-
-    // TODO: move these into protected
-    // more on private side
-    void updateSearch(const NodeSet &successors);
-    void updateGraph(const Node &parent,
-                     const NodeSet &successors);
-    Node getProminentSuccessor(const NodeSet &successors) const;
-    bool predExists(const Node &node) const;
-    bool nodeWorthExpanding(const Node &node) const;
-    double computeAveragePathCost() const;
-
-    bool isLost(int N, double perc) const;
-
-    void visualize() const;
-
-   private:
-    int _querySize = 0;
-    int _slidingWindowSize = 5;  // frames
-    bool _needReloc = false;
-    double _expansionRate = -1.0;
-    double _nonMatchCost = -1.0;
-
-    std::priority_queue<Node> _frontier;
-    // stores parent for each node
-    PredMap _pred;
-    // stores the accumulative  cost for each node
-    AccCostsMap _accCosts;
-    Node _currentBestHyp;
-
-    SuccessorManager::Ptr _successorManager = nullptr;
-    iLocVisualizer::Ptr _vis = nullptr;
-
-    NodeSet _expandedRecently;
+public:
+  using PredMap = std::unordered_map<int, std::unordered_map<int, Node>>;
+  using AccCostsMap = std::unordered_map<int, std::unordered_map<int, double>>;
+
+  OnlineLocalizer();
+  ~OnlineLocalizer() {}
+  void setQuerySize(int size) { _querySize = size; }
+  bool setSuccessorManager(SuccessorManager *successorManager);
+  bool setVisualizer(iLocVisualizer::Ptr vis);
+  bool setExpansionRate(double rate);
+  bool setNonMatchingCost(double non_match);
+
+  /**
+   * @brief      dumps path to the file. Line format: quId refId status (0-
+   * hidden, 1-real)
+   *
+   * @param[in]  filename  The filename
+   */
+  void printPath(const std::string &filename) const;
+
+  bool isReady() const;
+  void run();
+  void processImage(int quId);
+  // core working function
+  void matchImage(int quId);
+  std::vector<PathElement> getCurrentPath() const;
+  std::vector<PathElement> getLastNmatches(int N) const;
+
+  void writeOutExpanded(const std::string &filename) const;
+
+  // TODO: move these into protected
+  // more on private side
+  void updateSearch(const NodeSet &successors);
+  void updateGraph(const Node &parent, const NodeSet &successors);
+  Node getProminentSuccessor(const NodeSet &successors) const;
+  bool predExists(const Node &node) const;
+  bool nodeWorthExpanding(const Node &node) const;
+  double computeAveragePathCost() const;
+
+  bool isLost(int N, double perc) const;
+
+  void visualize() const;
+
+private:
+  int _querySize = 0;
+  int _slidingWindowSize = 5; // frames
+  bool _needReloc = false;
+  double _expansionRate = -1.0;
+  double _nonMatchCost = -1.0;
+
+  std::priority_queue<Node> _frontier;
```

Comment:
nit: you seemed to have changed the names for the other members to have a trailing `_` instead of the leading one. You could also do it here. But feel free to postpone this too.

Reply:
Yeah, I know. Will make a bigger refactoring for the whole `online_localizer` and then change the variables to the new names all together. Thanks.